spec/controllers/contact/govuk/accessible_format_requests_controller_spec.rb

```diff
@@ -93,6 +93,62 @@
         expect(response.body).to have_css("form#request-accessible-format[action=\"/contact/govuk/request-accessible-format/sent\"]")
       end
     end
+
+    context "error messages" do
+      let(:required_inputs) { Hash[required_keys.collect { |v| [v.to_sym, example_param] }] }
+      context "questions with required inputs" do
+        let(:required_input_question) { format_request_questions.detect { |question| question["inputs"][0]["error_message"] } }
+        let(:required_input_question_number) { format_request_questions.find_index(required_input_question) + 1 }
+        describe "when a required input is not entered" do
+          before { post :form, params: base_params.merge({ 'question_number': required_input_question_number + 1, 'previous_question_number': required_input_question_number }) }
+          it "reposts a form requesting the previous question" do
+            expect(response.body).to have_css("input[name=question_number][value=#{required_input_question_number}]", visible: false)
+          end
```

Comment:
There's a lot to unpack here! Newlines before each new indentation block (`context`, `before` etc) would help somewhat. I also wonder if some of these `let`s can be moved inline inside the `it` block, as standard variables, which should make this easier to mentally parse.

Reply:
I've attempted to tidy this, it is complicated! Adding line breaks and some comments, but unfortunately the `lets` are already in the last possible point without causing duplication, as they are used in multiple tests